### PR TITLE
KAFKA-9739:  Fixes null key changing child node

### PR DESCRIPTION
**Pull request by bbejeck**
2.4 port of #8400 since cherry-picking not possible

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
@vvcephei can you take a look this? It's identical to #8400, just for 2.4 branch.  Once merged I'll cherry-pick this back.

**Review by vvcephei — APPROVED** [review]
LGTM! (Pending the tests ;) )

Thanks again, @bbejeck

**Comment by bbejeck** [comment]
merged #8416 into 2.4

**Pull request merged**
